ballerina-tests/http-service-tests/tests/enum_types_for_query_param_tests.bal

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+
+public enum Status {
+    OLD,
+    NEW
+}
+
+service /enumQueryParam on generalListener {
+    resource function get album(Status status) returns string {
+        if status is OLD {
+            return "Old album";
+        }
+        return "New album";
+    }
+
+    resource function get song(Status[] status) returns string {
+        if status[0] is OLD && status[1] is NEW {
+            return "Old & New Song";
+        }
+        return "Unknown Song";
+    }
+}
+
+@test:Config {}
+function testEnumTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/album?status=OLD");
+    test:assertEquals(response, "Old album");
+}
+
+@test:Config {}
+function testEnumArrayTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/song?status=OLD&status=NEW");
+    test:assertEquals(response, "Old & New Song");
+}
```

Comment:
Can we also add some negative tests by passing some string values which are not compatible with the enum type?

Reply:
Updated with some changes. Older implementation didnt give errors with invalid enums